File: app/src/main/java/knf/kuma/videoservers/IzanagiServer.java
package knf.kuma.videoservers;

import android.content.Context;

import org.json.JSONObject;
import org.jsoup.Jsoup;

import androidx.annotation.Nullable;
import knf.kuma.commons.BypassUtil;

import static knf.kuma.videoservers.VideoServer.Names.IZANAGI;

public class IzanagiServer extends Server {
    public IzanagiServer(Context context, String baseLink) {
        super(context, baseLink);
    }

    @Override
    public boolean isValid() {
        return baseLink.contains("s=izanagi");
    }

    @Override
    public String getName() {
        return IZANAGI;
    }

    @Nullable
    @Override
    public VideoServer getVideoServer() {
        String frame = baseLink.substring(baseLink.indexOf("'") + 1, baseLink.lastIndexOf("'"));
        String down_link = Jsoup.parse(frame).select("iframe").first().attr("src");
        try {
            String link = new JSONObject(Jsoup.connect(down_link.replace("embed", "check")).cookies(BypassUtil.getMapCookie(context)).userAgent(BypassUtil.userAgent).get().body().text()).getString("file").replace("\\", "");
            link = link.replaceAll("/", "//").replace(":////", "://");
            return new VideoServer(IZANAGI, new Option(getName(), null, link));
        } catch (Exception e) {
            return null;
        }
    }
}

File: app/src/main/java/knf/kuma/videoservers/IzanagiServer.kt
package knf.kuma.videoservers

import android.content.Context
import knf.kuma.commons.PatternUtil
import knf.kuma.commons.jsoupCookies
import knf.kuma.videoservers.VideoServer.Names.IZANAGI
import org.json.JSONObject

class IzanagiServer(context: Context, baseLink: String) : Server(context, baseLink) {

    override val isValid: Boolean
        get() = baseLink.contains("s=izanagi")

    override val name: String
        get() = IZANAGI

    override val videoServer: VideoServer?
        get() {
            val downLink = PatternUtil.extractLink(baseLink)
            return try {
                val link = JSONObject(jsoupCookies(downLink.replace("embed", "check")).get().body().text()).getString("file").replace("\\", "")
                VideoServer(IZANAGI,
                        mutableListOf(
                                Option(name, null, link),
                                Option(name, null, link.replace("/".toRegex(), "//").replace(":////", "://"))
                        ))
            } catch (e: Exception) {
                null
            }

        }
}
